c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -603,6 +735,13 @@ class DeviceCachingAllocator {
     } else if (&pool == &large_blocks) {
       return StatType::LARGE_POOL;
     } else {
+      for (auto& gp : graph_pools) {
```

Comment:
This changes the asymptotics of this function for graph pools, would be good to make sure people aren't assuming this is free. (It might be easier to just store StatType on BlockPool post this change)

Reply:
hmm im not sure i understand what you're suggesting here.

Reply:
The potential problem is a call site calling this on private pools expecting it to be O(1) when it is not. The potential solution is if a pool records, as a field, whether or not it is small or large, you don't have to do the pointer test; you can just read it out of the field. But this is only a very minor problem.

Reply:
Augmented BlockPool with `is_small` and `owner_PrivatePool` fields to avoid these address-checking loops.